sklearn/manifold/_barnes_hut_tsne.pyx

```diff
@@ -436,13 +436,14 @@ cdef long count_points(Node* root, long count) nogil:
         # one point, and then the other neighboring cells
         # don't get filled in
     if DEBUGFLAG:
-        printf("[t-SNE] %i points in this node\n", count)
+        printf("[t-SNE] %li points in this node\n", count)
     return count
 
 
-cdef float compute_gradient(float[:,:] val_P,
+cdef float compute_gradient(float[:] val_P,
                             float[:,:] pos_reference,
-                            np.int64_t[:,:] neighbors,
+                            np.int64_t[:] neighbors,
+                            np.int64_t[:] indptr,
```

Comment:
This makes it easy to efficiently symmetrize the `conditional_P` matrix using scipy operation for sparse matrices.
Symmetrization is done in the [reference implementation](https://github.com/lvdmaaten/bhtsne) so I tried to be as close as possible to it. Although I haven't reviewed it all yet.